In [33]:
import pandas as pd
import numpy as np

main = pd.read_csv('players.csv')
midstats = pd.read_csv('miscplay.csv')
shooting = pd.read_csv('shooting.csv')
merged = pd.merge(main, midstats, on=["Rk", "Player", "Nation", "Pos", "Squad", "Comp", "Age", "Born", "Matches", "-9999", "CrdY", "CrdR", "90s"], how='left')
merged = pd.merge(merged, shooting, on=["Rk", "Player", "Nation", "Pos", "Squad", "Comp", "Age", "Born", "Matches", "-9999", "Gls","PK", "PKatt","90s"], how='left')
merged = merged.drop(columns = ["Gls.1", "Ast.1", "G+A.1", "G-PK.1", "Rk", "-9999"])
merged.head()
merged.to_csv('merged.csv', index=False)



Merging all different player stat tables and cleaning data.

In [34]:
main = main[main["Min"] >= 900]
main = main[main["Age"] <= 23]
main = main[main["Pos"] == "MF"]
main.info()

<class 'pandas.DataFrame'>
Index: 172 entries, 41 to 2806
Data columns (total 27 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Rk       172 non-null    int64  
 1   Player   172 non-null    str    
 2   Nation   172 non-null    str    
 3   Pos      172 non-null    str    
 4   Squad    172 non-null    str    
 5   Comp     172 non-null    str    
 6   Age      172 non-null    float64
 7   Born     172 non-null    float64
 8   MP       172 non-null    int64  
 9   Starts   172 non-null    int64  
 10  Min      172 non-null    int64  
 11  90s      172 non-null    float64
 12  Gls      172 non-null    int64  
 13  Ast      172 non-null    int64  
 14  G+A      172 non-null    int64  
 15  G-PK     172 non-null    int64  
 16  PK       172 non-null    int64  
 17  PKatt    172 non-null    int64  
 18  CrdY     172 non-null    int64  
 19  CrdR     172 non-null    int64  
 20  Gls.1    172 non-null    float64
 21  Ast.1    172 non-null    float

In [35]:
merged = merged[merged["Min"] >= 900]
merged = merged[merged["Age"] <= 23]
merged = merged[merged["Pos"] == "MF"]
merged.info()
merged.head()

<class 'pandas.DataFrame'>
Index: 172 entries, 41 to 2806
Data columns (total 38 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Player   172 non-null    str    
 1   Nation   172 non-null    str    
 2   Pos      172 non-null    str    
 3   Squad    172 non-null    str    
 4   Comp     172 non-null    str    
 5   Age      172 non-null    float64
 6   Born     172 non-null    float64
 7   MP       172 non-null    int64  
 8   Starts   172 non-null    int64  
 9   Min      172 non-null    int64  
 10  90s      172 non-null    float64
 11  Gls      172 non-null    int64  
 12  Ast      172 non-null    int64  
 13  G+A      172 non-null    int64  
 14  G-PK     172 non-null    int64  
 15  PK       172 non-null    int64  
 16  PKatt    172 non-null    int64  
 17  CrdY     172 non-null    int64  
 18  CrdR     172 non-null    int64  
 19  G+A-PK   172 non-null    float64
 20  Matches  172 non-null    str    
 21  2CrdY    172 non-null    int64

,Player,Nation,Pos,Squad,Comp,Age,Born,MP,Starts,Min,...,PKwon,PKcon,OG,Sh,SoT,SoT%,Sh/90,SoT/90,G/Sh,G/SoT
41,Lucien Agoume,fr FRA,MF,Sevilla,es La Liga,23.0,2002.0,34,31,2686,...,NaN,NaN,0,15.0,2.0,13.3,0.50,0.07,0.07,0.50
65,Ebenezer Akinsanmiro,ng NGA,MF,Pisa,it Serie A,20.0,2004.0,24,16,1444,...,NaN,NaN,0,13.0,3.0,23.1,0.81,0.19,0.00,0.00
91,Sergi Altimira,es ESP,MF,Real Betis,es La Liga,23.0,2001.0,24,13,1186,...,NaN,NaN,0,7.0,3.0,42.9,0.53,0.23,0.14,0.33
92,Carlos Álvarez,es ESP,MF,Levante,es La Liga,21.0,2003.0,30,23,1890,...,NaN,NaN,0,30.0,11.0,36.7,1.43,0.52,0.10,0.27
113,Elliot Anderson,eng ENG,MF,Nottingham,eng Premier League,22.0,2002.0,38,37,3331,...,NaN,NaN,0,44.0,10.0,22.7,1.19,0.27,0.09,0.40


merged.info() and main.info() check that no player has been lost in the merging process since both show 172 entries.

After having merged and cleaned the data for the player standard stats, miscellaneous stats and shooting stats, I ran into a problem. There is no available data from Fbref on categories such as defence and possession meaning I will have to use a different dataset to obtain these statistics and acquire a full midfielder profile.